client/state/notices/middleware.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+import { truncate } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { successNotice, errorNotice } from 'state/notices/actions';
+import { getSitePost } from 'state/posts/selectors';
+import {
+	POST_DELETE_FAILURE,
+	POST_DELETE_SUCCESS,
+	POST_SAVE_SUCCESS
+} from 'state/action-types';
+
+export const OBSERVERS = {
+	[ POST_DELETE_FAILURE ]: ( action, dispatch, getState ) => {
```

Comment:
>the stuff that's in there now has no bearing on the reader, but it's being included for the reader bundle

I wonder if this could be an argument against keeping them alongside the action creators. It seems sensible to me that the notices behavior should always be globally aware of action types it's concerned with, not dependent on the action creators file having been loaded beforehand (though presumably in most/all cases the action creator to have triggered the action will most likely have been loaded beforehand anyways).

>javascript decorators are still stage-1

I've a love/hate relationship with decorators. They seem like a neat idea, but different enough that I worry it's the type of proposal that won't ever be adopted, and would be painful to rip out of code if we were to start depending on it now. I'd be more comfortable waiting until it at least reaches stage 2.